pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -571,10 +587,23 @@ func (p *staticPolicy) generateCPUTopologyHints(availableCPUs cpuset.CPUSet, reu
 	// to the minAffinitySize. Only those with an equal number of bits set (and
 	// with a minimal set of numa nodes) will be considered preferred.
 	for i := range hints {
+		if p.options.AlignBySocket && p.isHintSocketAligned(hints[i].NUMANodeAffinity) {
+			hints[i].Preferred = true
+			continue
+		}
 		if hints[i].NUMANodeAffinity.Count() == minAffinitySize {
 			hints[i].Preferred = true
 		}
 	}
 
 	return hints
 }
+
+//isHintSocketAligned function return true if numa nodes in hint are within one socket.
+func (p *staticPolicy) isHintSocketAligned(hint bitmask.BitMask) bool {
```

Comment:
Given the name it seems this should take a hint as a paramater, not a bitmask.

Reply:
fixed function input parameter